common/src/stack/command/stack/commands/__init__.py

```diff
@@ -1398,12 +1398,10 @@ def getHostname(self, hostname=None, subnet=None):
 						try:
 							name = '%s.%s' % (hostname, domain)
 							addr = socket.gethostbyname(name)
-							hostname = name
-							break
+							if addr:
+								return self.getHostname(name)
 						except:
 							pass
-					if addr:
-						return self.getHostname(hostname)
 
 					fin.close()
```

Comment:
Out of scope: This file appears to not be closed in the case where addr is true.

Reply:
It will close, once the variable is garbage collected. It would be cleaner to open the file with a `with` context manager, but I just wanted to quickly test if this change would fix the problem.